spec/solver_spec.rb

```diff
@@ -0,0 +1,43 @@
+require_relative '../solver'
+
+describe Solver do
+  before(:all) do
+    @solver = Solver.new
+  end
+
+  describe '#factorial' do
+    it 'should return argument error if num is negative' do
+      expect { @solver.factorial(-2) }.to raise_error(ArgumentError)
+    end
+    it 'should return 1 when input argument is 1' do
+      expect(@solver.factorial(1)).to eql(1)
+    end
+    it 'should return 120 when input argument is 5' do
+      expect(@solver.factorial(5)).to eq(120)
+    end
```

Comment:
- [ ] As required for the project, kindly make sure you also test the `factorial()` when the input is **0** to get the correct answer.